### PR TITLE
fix(selector): report CSS-load and persistence failures from applyTheme

### LINKED ISSUE #695: fix(selector): report CSS-load and persistence failures from applyTheme
## Summary

Greptile review of #692 (integration API) flagged that `applyTheme` returns `true` and emits `turbo-theme-change` even when (a) theme CSS loading silently fails (`loadThemeCSS` swallows load errors internally) or (b) `saveTheme` returns `false` (localStorage unavailable/quota). The API should be honest about partial failures.

## Problem

Consumers cannot distinguish a fully applied theme (attributes + CSS + persistence) from a partial application where only DOM attributes changed.

## Proposed Solution

- Propagate CSS-load failure out of `applyThemeToDocument` (without reintroducing FOUC behavior changes for the dropdown path)
- Check `saveTheme`'s return value; log and reflect persistence failure in the result
- Either return a richer result object (`{ applied, cssLoaded, persisted }`) or return `false` on partial failure — pick one and document it
- Update integration tests accordingly

## References

- Greptile threads on #692 (integration.ts P1/P2)
- Depends on #680 landing (PR #692)

### PR DESCRIPTION
**Pull request by TurboCoder13**
## Summary

`applyTheme` in the theme-selector integration API previously returned `true` and dispatched `turbo-theme-change` even when the theme CSS failed to load or persistence to localStorage failed. This PR makes every stage of theme application honestly reported, per #695 and its two scope-extension comments.

- **Integration API**: `applyTheme` now returns an `ApplyThemeResult` object — `{ applied, cssLoaded, persisted }` — reporting each stage independently. `saveTheme`'s return value is checked (`persisted`), and CSS-load failure propagates out of `applyThemeToDocument` (`cssLoaded`). The `turbo-theme-change` event fires only when the theme is visually applied (`applied && cssLoaded`); persistence failure alone does not suppress it, since the visual state did change.
- **Adoption-path awaiting**: when `loadThemeCSS` adopts the blocking script's `CSS_LINK_ID` link, the repoint still happens synchronously (FOUC-avoidance intact), but load/error handlers are now attached to the repointed link and awaited. On failure the link is rolled back to its prior identity and href, so the previously working stylesheet stays in effect and a later switch can retry. When the blocking link already points at the requested stylesheet, no refetch is awaited.
- **Shared link-ID builder**: `themeLinkId(themeId)` is exported from `theme-loader.ts` and used by both the loader and `lazy-css.ts`, removing the duplicated `theme-${id}-css` convention.
- `loadThemeCSSOnDemand` now consumes `loadThemeCSS`'s boolean result directly instead of a post-await DOM re-check that resolved `true` before the fetch settled.

## Chosen reporting shape and why

`{ applied, cssLoaded, persisted }` (the richer object recommended in the issue) rather than `false` on partial failure:

- A bare boolean cannot distinguish "nothing happened" (invalid ID) from "attributes changed but CSS or persistence failed", which was the core complaint.
- The only in-repo caller (the dropdown's `onThemeSelect`) ignores the return value, so the richer shape breaks nothing.
- Keeping boolean-ish truthiness out of the API prevents `if (await applyTheme(...))` from silently conflating partial failure with rejection.

## Breaking-change note

The integration API merged in #692 **after** v0.36.2 and has not shipped in any release (no tag contains it), so changing the return type from `Promise<boolean>` to `Promise<ApplyThemeResult>` is not a published breaking change.

## Bundles and budget

Committed regenerated bundles (`bun run build:js`): `assets/js/theme-selector.js`, `.js.map`, `.min.js`. `packages/theme-selector/dist/index.js` is ~123.6KB against the 128KB budget in `test/integration/bundle-size.test.ts` — no budget bump needed.

## Testing

- 14 new theme-selector tests (409 in-package, up from 395): partial-failure reporting for CSS-load and persistence failure, event suppression on CSS failure, adoption-path error propagation (driving the repointed link's `onerror`), adoption rollback, settlement awaiting, same-href short-circuit, and the shared `themeLinkId` builder. Typed test doubles, no `as any` DOM mocks.
- One pre-existing mock-based test (`test/public-api/edge-cases/base-url-paths.test.ts`) had a `getElementById` override that returned a generic mock for the blocking link ID; it now returns `null` there, matching the `createBaseUrlTestSetup` helper it overrides.
- Full suite green: 3254 tests / 96 files; `lintro chk` 0 issues.

Closes #695

<details><summary><h3>Greptile Summary</h3></summary>

This PR makes `applyTheme` in the integration API honest about per-stage outcomes, returning `ApplyThemeResult { applied, cssLoaded, persisted }` instead of a plain boolean. The `turbo-theme-change` event is now gated on CSS load success, the blocking-script link adoption path gains load/error observability with rollback on failure, and `themeLinkId` is promoted to a shared export eliminating the duplicated `theme-${id}-css` convention.

- `loadThemeCSS` is refactored into `adoptBlockingLink` (handles FOUC-safe repoint + awaited settlement + rollback) and a simpler fresh-link path, both returning `boolean`.
- `saveTheme`'s return value is captured as `persisted`; CSS load failure suppresses the `turbo-theme-change` event but not localStorage persistence.
- 14 new in-package tests cover partial-failure reporting, adoption-path error propagation, rollback, settlement awaiting, and the same-href short-circuit; the pre-existing `base-url-paths` test is fixed to return `null` for the blocking-link ID it never intended to exercise.
</details>

<h3>Confidence Score: 4/5</h3>

Safe to merge with the two logic notes addressed; the core refactor is well-structured and thoroughly tested.

The adoption-path rollback leaves the blocking link's href unreset when the original link had no href attribute, and saveTheme runs before CSS load is confirmed — meaning a CSS failure leaves localStorage pointing at a broken theme that the blocking script will retry on every subsequent page load. Both are edge-case situations unlikely to hit users today, but the href rollback gap is a straightforward correctness bug in the new code path. Everything else — result shape, event suppression, test coverage, the shared themeLinkId export — is clean.

packages/theme-selector/src/theme-loader.ts (adoptBlockingLink rollback) and packages/theme-selector/src/integration.ts (persistence-before-CSS-load ordering).

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/theme-selector/src/theme-loader.ts | Core refactor: `loadThemeCSS` now returns `boolean`, adopts blocking link via new `adoptBlockingLink` helper with rollback on failure, and exports `themeLinkId` as shared SSoT. One edge case in the rollback path when `previousHref` is null. |
| packages/theme-selector/src/integration.ts | Integration `applyTheme` returns `ApplyThemeResult` with per-stage fields; event dispatch gated on `cssLoaded`; `saveTheme` is called before CSS load confirmation, meaning localStorage is updated even when CSS subsequently fails. |
| packages/theme-selector/src/apply-theme.ts | Minimal change: propagates `loadThemeCSS` boolean result through to the caller; JSDoc updated to reflect new return semantics. |
| packages/theme-selector/src/lazy-css.ts | Removes local `themeCSSLinkId` in favour of imported `themeLinkId`; `loadThemeCSSOnDemand` now directly returns `loadThemeCSS`'s boolean instead of a post-await DOM re-check. |
| packages/theme-selector/test/theme-loader.test.ts | 14 new tests covering boolean return, adoption-path failure/rollback/settlement, same-href short-circuit, and the shared `themeLinkId` builder. |
| packages/theme-selector/test/integration.test.ts | Updated assertions for `ApplyThemeResult` shape; new describe block covers persistence failure, CSS failure, and event suppression on CSS failure. |
| test/public-api/edge-cases/base-url-paths.test.ts | One-line fix: `getElementById('turbo-theme-css')` now returns `null` to match the pre-existing `createBaseUrlTestSetup` helper, preventing the blocking-link adoption path from being exercised in unrelated tests. |

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant Caller
    participant integration.ts as integration::applyTheme
    participant storage.ts as saveTheme
    participant apply-theme.ts as applyThemeToDocument
    participant theme-loader.ts as loadThemeCSS
    participant adoptBlockingLink

    Caller->>integration.ts: applyTheme(themeId)
    integration.ts->>integration.ts: isValidThemeId + validIds.has check
    alt invalid theme
        integration.ts-->>Caller: "{applied:false, cssLoaded:false, persisted:false}"
    else valid theme
        integration.ts->>storage.ts: saveTheme(windowObj, themeId, validIds)
        storage.ts-->>integration.ts: persisted (boolean)
        integration.ts->>apply-theme.ts: applyThemeToDocument(doc, themeId)
        apply-theme.ts->>apply-theme.ts: resolveTheme(themeId)
        apply-theme.ts->>apply-theme.ts: applyThemeClass(doc, themeId)
        apply-theme.ts->>theme-loader.ts: loadThemeCSS(doc, theme, baseUrl)
        alt blocking link present (adoption path)
            theme-loader.ts->>adoptBlockingLink: adoptBlockingLink(doc, link, theme, baseUrl)
            adoptBlockingLink->>adoptBlockingLink: set link.id, data-theme-id
            alt href already matches
                adoptBlockingLink-->>theme-loader.ts: true (no refetch)
            else href differs
                adoptBlockingLink->>adoptBlockingLink: "link.href = resolvedHref (sync, FOUC-safe)"
                adoptBlockingLink->>adoptBlockingLink: await loadCSSWithTimeout(link)
                alt load succeeds
                    adoptBlockingLink-->>theme-loader.ts: true
                else load fails
                    adoptBlockingLink->>adoptBlockingLink: rollback id, data-theme-id, href
                    adoptBlockingLink-->>theme-loader.ts: false
                end
            end
        else no blocking link
            theme-loader.ts->>theme-loader.ts: create new link, await loadCSSWithTimeout
            theme-loader.ts-->>apply-theme.ts: true/false
        end
        apply-theme.ts-->>integration.ts: cssLoaded (boolean)
        alt cssLoaded
            integration.ts->>integration.ts: emitThemeChange(doc, themeId)
        end
        integration.ts-->>Caller: "{applied:true, cssLoaded, persisted}"
    end
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant Caller
    participant integration.ts as integration::applyTheme
    participant storage.ts as saveTheme
    participant apply-theme.ts as applyThemeToDocument
    participant theme-loader.ts as loadThemeCSS
    participant adoptBlockingLink

    Caller->>integration.ts: applyTheme(themeId)
    integration.ts->>integration.ts: isValidThemeId + validIds.has check
    alt invalid theme
        integration.ts-->>Caller: "{applied:false, cssLoaded:false, persisted:false}"
    else valid theme
        integration.ts->>storage.ts: saveTheme(windowObj, themeId, validIds)
        storage.ts-->>integration.ts: persisted (boolean)
        integration.ts->>apply-theme.ts: applyThemeToDocument(doc, themeId)
        apply-theme.ts->>apply-theme.ts: resolveTheme(themeId)
        apply-theme.ts->>apply-theme.ts: applyThemeClass(doc, themeId)
        apply-theme.ts->>theme-loader.ts: loadThemeCSS(doc, theme, baseUrl)
        alt blocking link present (adoption path)
            theme-loader.ts->>adoptBlockingLink: adoptBlockingLink(doc, link, theme, baseUrl)
            adoptBlockingLink->>adoptBlockingLink: set link.id, data-theme-id
            alt href already matches
                adoptBlockingLink-->>theme-loader.ts: true (no refetch)
            else href differs
                adoptBlockingLink->>adoptBlockingLink: "link.href = resolvedHref (sync, FOUC-safe)"
                adoptBlockingLink->>adoptBlockingLink: await loadCSSWithTimeout(link)
                alt load succeeds
                    adoptBlockingLink-->>theme-loader.ts: true
                else load fails
                    adoptBlockingLink->>adoptBlockingLink: rollback id, data-theme-id, href
                    adoptBlockingLink-->>theme-loader.ts: false
                end
            end
        else no blocking link
            theme-loader.ts->>theme-loader.ts: create new link, await loadCSSWithTimeout
            theme-loader.ts-->>apply-theme.ts: true/false
        end
        apply-theme.ts-->>integration.ts: cssLoaded (boolean)
        alt cssLoaded
            integration.ts->>integration.ts: emitThemeChange(doc, themeId)
        end
        integration.ts-->>Caller: "{applied:true, cssLoaded, persisted}"
    end
```

</a>
</details>

<a href="https://app.greptile.com/api/ide/cursor?prompt=Fix%20the%20following%203%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Ftheme-loader.ts%3A216-218%0AIncomplete%20%60href%60%20rollback%20when%20the%20blocking%20link%20had%20no%20%60href%60%20attribute.%20After%20%60blockingLink.href%20%3D%20resolvedHref%60%20is%20set%20at%20line%20208%2C%20a%20load%20failure%20should%20restore%20the%20attribute%20to%20its%20original%20state.%20When%20%60previousHref%60%20is%20%60null%60%20the%20attribute%20was%20absent%2C%20but%20the%20guard%20skips%20the%20restore%20branch%20entirely%20%E2%80%94%20leaving%20%60CSS_LINK_ID%60%20pointing%20at%20the%20failed%20theme%20URL%20instead%20of%20being%20href-less.%20The%20fix%20is%20to%20remove%20the%20attribute%20in%20the%20%60else%60%20branch.%0A%0A%60%60%60suggestion%0A%20%20%20%20if%20%28previousHref%20!%3D%3D%20null%29%20%7B%0A%20%20%20%20%20%20blockingLink.href%20%3D%20previousHref%3B%0A%20%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20%20blockingLink.removeAttribute%28'href'%29%3B%0A%20%20%20%20%7D%0A%60%60%60%0A%0A%23%23%23%20Issue%202%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Fintegration.ts%3A116-121%0A**%60saveTheme%60%20runs%20before%20CSS%20load%20is%20confirmed**%0A%0A%60saveTheme%60%20is%20called%20unconditionally%20before%20%60applyThemeToDocument%60%20awaits%20the%20network.%20When%20CSS%20fails%20%28%60cssLoaded%3A%20false%60%29%2C%20localStorage%20already%20reflects%20the%20new%20theme%20ID.%20On%20the%20next%20page%20load%20the%20blocking%20script%20will%20attempt%20to%20fetch%20the%20same%20%28possibly%20permanently%20broken%29%20stylesheet%2C%20setting%20up%20a%20cycle%20of%20failures%20that%20is%20never%20auto-cleared.%20A%20caller%20observing%20%60cssLoaded%3A%20false%60%20would%20need%20to%20manually%20call%20%60saveTheme%60%20again%20with%20the%20old%20ID%20to%20repair%20the%20stored%20preference.%20Consider%20whether%20persistence%20should%20be%20deferred%20until%20%60cssLoaded%60%20is%20confirmed%2C%20or%20at%20least%20document%20the%20chosen%20ordering%20explicitly%20in%20the%20JSDoc%20so%20callers%20understand%20what%20a%20%60cssLoaded%3A%20false%60%20result%20leaves%20in%20storage.%0A%0A%23%23%23%20Issue%203%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Fintegration.ts%3A57-60%0AStray%20double%20blank%20line%20between%20the%20%60ApplyThemeResult%60%20interface%20closing%20brace%20and%20the%20%60Unsubscribe%60%20type%20alias%20%E2%80%94%20single%20blank%20line%20is%20the%20convention%20used%20elsewhere%20in%20the%20file.%0A%0A%60%60%60suggestion%0A%7D%0A%0A%2F**%20Removes%20a%20listener%20previously%20registered%20via%20subscribeToThemeChanges.%20*%2F%0A%60%60%60%0A%0A&pr=698&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursorDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursor.svg?v=6">Fix All in Cursor</picture></a> <a href="https://app.greptile.com/ide/claude-code?prompt=Fix%20the%20following%203%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Ftheme-loader.ts%3A216-218%0AIncomplete%20%60href%60%20rollback%20when%20the%20blocking%20link%20had%20no%20%60href%60%20attribute.%20After%20%60blockingLink.href%20%3D%20resolvedHref%60%20is%20set%20at%20line%20208%2C%20a%20load%20failure%20should%20restore%20the%20attribute%20to%20its%20original%20state.%20When%20%60previousHref%60%20is%20%60null%60%20the%20attribute%20was%20absent%2C%20but%20the%20guard%20skips%20the%20restore%20branch%20entirely%20%E2%80%94%20leaving%20%60CSS_LINK_ID%60%20pointing%20at%20the%20failed%20theme%20URL%20instead%20of%20being%20href-less.%20The%20fix%20is%20to%20remove%20the%20attribute%20in%20the%20%60else%60%20branch.%0A%0A%60%60%60suggestion%0A%20%20%20%20if%20%28previousHref%20!%3D%3D%20null%29%20%7B%0A%20%20%20%20%20%20blockingLink.href%20%3D%20previousHref%3B%0A%20%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20%20blockingLink.removeAttribute%28'href'%29%3B%0A%20%20%20%20%7D%0A%60%60%60%0A%0A%23%23%23%20Issue%202%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Fintegration.ts%3A116-121%0A**%60saveTheme%60%20runs%20before%20CSS%20load%20is%20confirmed**%0A%0A%60saveTheme%60%20is%20called%20unconditionally%20before%20%60applyThemeToDocument%60%20awaits%20the%20network.%20When%20CSS%20fails%20%28%60cssLoaded%3A%20false%60%29%2C%20localStorage%20already%20reflects%20the%20new%20theme%20ID.%20On%20the%20next%20page%20load%20the%20blocking%20script%20will%20attempt%20to%20fetch%20the%20same%20%28possibly%20permanently%20broken%29%20stylesheet%2C%20setting%20up%20a%20cycle%20of%20failures%20that%20is%20never%20auto-cleared.%20A%20caller%20observing%20%60cssLoaded%3A%20false%60%20would%20need%20to%20manually%20call%20%60saveTheme%60%20again%20with%20the%20old%20ID%20to%20repair%20the%20stored%20preference.%20Consider%20whether%20persistence%20should%20be%20deferred%20until%20%60cssLoaded%60%20is%20confirmed%2C%20or%20at%20least%20document%20the%20chosen%20ordering%20explicitly%20in%20the%20JSDoc%20so%20callers%20understand%20what%20a%20%60cssLoaded%3A%20false%60%20result%20leaves%20in%20storage.%0A%0A%23%23%23%20Issue%203%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Fintegration.ts%3A57-60%0AStray%20double%20blank%20line%20between%20the%20%60ApplyThemeResult%60%20interface%20closing%20brace%20and%20the%20%60Unsubscribe%60%20type%20alias%20%E2%80%94%20single%20blank%20line%20is%20the%20convention%20used%20elsewhere%20in%20the%20file.%0A%0A%60%60%60suggestion%0A%7D%0A%0A%2F**%20Removes%20a%20listener%20previously%20registered%20via%20subscribeToThemeChanges.%20*%2F%0A%60%60%60%0A%0A&repo=lgtm-hq%2Fturbo-themes&pr=698&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=6">Fix All in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/codex?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22lgtm-hq%2Fturbo-themes%22%20on%20the%20existing%20branch%20%22fix%2F695-selector-partial-failure-reporting%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22fix%2F695-selector-partial-failure-reporting%22.%0A%0AFix%20the%20following%203%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Ftheme-loader.ts%3A216-218%0AIncomplete%20%60href%60%20rollback%20when%20the%20blocking%20link%20had%20no%20%60href%60%20attribute.%20After%20%60blockingLink.href%20%3D%20resolvedHref%60%20is%20set%20at%20line%20208%2C%20a%20load%20failure%20should%20restore%20the%20attribute%20to%20its%20original%20state.%20When%20%60previousHref%60%20is%20%60null%60%20the%20attribute%20was%20absent%2C%20but%20the%20guard%20skips%20the%20restore%20branch%20entirely%20%E2%80%94%20leaving%20%60CSS_LINK_ID%60%20pointing%20at%20the%20failed%20theme%20URL%20instead%20of%20being%20href-less.%20The%20fix%20is%20to%20remove%20the%20attribute%20in%20the%20%60else%60%20branch.%0A%0A%60%60%60suggestion%0A%20%20%20%20if%20%28previousHref%20!%3D%3D%20null%29%20%7B%0A%20%20%20%20%20%20blockingLink.href%20%3D%20previousHref%3B%0A%20%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20%20blockingLink.removeAttribute%28'href'%29%3B%0A%20%20%20%20%7D%0A%60%60%60%0A%0A%23%23%23%20Issue%202%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Fintegration.ts%3A116-121%0A**%60saveTheme%60%20runs%20before%20CSS%20load%20is%20confirmed**%0A%0A%60saveTheme%60%20is%20called%20unconditionally%20before%20%60applyThemeToDocument%60%20awaits%20the%20network.%20When%20CSS%20fails%20%28%60cssLoaded%3A%20false%60%29%2C%20localStorage%20already%20reflects%20the%20new%20theme%20ID.%20On%20the%20next%20page%20load%20the%20blocking%20script%20will%20attempt%20to%20fetch%20the%20same%20%28possibly%20permanently%20broken%29%20stylesheet%2C%20setting%20up%20a%20cycle%20of%20failures%20that%20is%20never%20auto-cleared.%20A%20caller%20observing%20%60cssLoaded%3A%20false%60%20would%20need%20to%20manually%20call%20%60saveTheme%60%20again%20with%20the%20old%20ID%20to%20repair%20the%20stored%20preference.%20Consider%20whether%20persistence%20should%20be%20deferred%20until%20%60cssLoaded%60%20is%20confirmed%2C%20or%20at%20least%20document%20the%20chosen%20ordering%20explicitly%20in%20the%20JSDoc%20so%20callers%20understand%20what%20a%20%60cssLoaded%3A%20false%60%20result%20leaves%20in%20storage.%0A%0A%23%23%23%20Issue%203%20of%203%0Apackages%2Ftheme-selector%2Fsrc%2Fintegration.ts%3A57-60%0AStray%20double%20blank%20line%20between%20the%20%60ApplyThemeResult%60%20interface%20closing%20brace%20and%20the%20%60Unsubscribe%60%20type%20alias%20%E2%80%94%20single%20blank%20line%20is%20the%20convention%20used%20elsewhere%20in%20the%20file.%0A%0A%60%60%60suggestion%0A%7D%0A%0A%2F**%20Removes%20a%20listener%20previously%20registered%20via%20subscribeToThemeChanges.%20*%2F%0A%60%60%60%0A%0A&repo=lgtm-hq%2Fturbo-themes&pr=698&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodexDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodex.svg?v=6">Fix All in Codex</picture></a>

<sub>Reviews (1): Last reviewed commit: ["chore(build): regenerate theme-selector ..."](https://github.com/lgtm-hq/turbo-themes/commit/529ecae38c4a671c8935d1ffe6d28c1d58714025) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=45638397)</sub>

> Greptile also left **3 inline comments** on this PR.

### CODE REVIEW

File: packages/theme-selector/src/theme-loader.ts [hunk 1]
```diff
@@ -143,75 +156,126 @@ export function applyThemeClass(doc: Document, themeId: string): void {
...
+  try {
+    await loadCSSWithTimeout(blockingLink, theme.id);
+  } catch (error) {
+    // Loading failed — restore the link's prior identity and href so the
+    // previously working stylesheet stays in effect.
+    blockingLink.id = CSS_LINK_ID;
+    blockingLink.removeAttribute('data-theme-id');
+    if (previousHref !== null) {
+      blockingLink.href = previousHref;
     }
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply packages/theme-selector/src/theme-loader.ts:218]: Valid edge case — when the blocking link shipped without an href, rollback should removeAttribute('href') rather than restore null-as-string. Deferred to #696 (noted there) since the no-href blocking link doesn't occur with the shipped blocking script.

File: packages/theme-selector/src/integration.ts [hunk 2]
```diff
@@ -57,29 +81,44 @@ function emitThemeChange(documentObj: Document, themeId: string): void {
...
-  saveTheme(windowObj, themeId, validIds);
-  await applyThemeToDocument(documentObj, themeId);
-  emitThemeChange(documentObj, themeId);
-  return true;
+  const persisted = saveTheme(windowObj, themeId, validIds);
+  const cssLoaded = await applyThemeToDocument(documentObj, themeId);
+  if (cssLoaded) {
+    emitThemeChange(documentObj, themeId);
+  }
+  return { applied: true, cssLoaded, persisted };
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply packages/theme-selector/src/integration.ts:121]: Intentional trade-off, keeping as-is: saveTheme persists the user's *intent*, so a transient network failure doesn't silently discard their choice — next-page-load retry of the persisted theme is desired behavior, and the ApplyThemeResult.persisted/cssLoaded fields give callers what they need to react differently if they want. Noted as a design decision on #696's discussion.

File: packages/theme-selector/src/integration.ts [hunk 3]
```diff
@@ -33,6 +33,30 @@ export interface ThemeChangeDetail {
...
+  cssLoaded: boolean;
+  /**
+   * True when the selection was persisted to localStorage. False when
+   * storage is unavailable (e.g. private browsing) or over quota.
+   */
+  persisted: boolean;
+}
+
+
 /** Removes a listener previously registered via subscribeToThemeChanges. */
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply packages/theme-selector/src/integration.ts:60]: Cosmetic, not flagged by lintro/prettier — folded into the #696 cleanup pass rather than a dedicated CI cycle here.
